workflows/metatdenovo.nf

```diff
@@ -73,17 +144,372 @@ workflow METATDENOVO {
+        } else {
+            ch_pe_reads_to_assembly = ch_interleaved.map { meta, fastq -> fastq }
+            ch_se_reads_to_assembly = Channel.empty()
+        }
+    }
+
+    //
+    // MODULE: Run Megahit or RNAspades on all interleaved fastq files
+    //
+    if ( params.assembly ) {
+        Channel
+            .value ( [ [ id: 'user_assembly' ], file(params.assembly) ] )
+            .set { ch_assembly_contigs }
+    } else if ( assembler == 'rnaspades' ) {
+        // 1. Write a yaml file for Spades
+        WRITESPADESYAML (
+            ch_pe_reads_to_assembly.toList(),
+            ch_se_reads_to_assembly.toList()
+        )
+        ch_versions    = ch_versions.mix(WRITESPADESYAML.out.versions)
+        // 2. Call the module with a channel with all fastq files plus the yaml
+        ch_pe_reads_to_assembly
+            .mix(ch_se_reads_to_assembly)
+            .collect()
+            .map { [ [ id:'rnaspades' ], it, [], [] ] }
+            .set { ch_spades }
+        SPADES (
+            ch_spades,
+            WRITESPADESYAML.out.yaml,
+            []
+        )
+        ch_assembly = SPADES.out.transcripts
+        ch_versions = ch_versions.mix(SPADES.out.versions)
+        FORMATSPADES( ch_assembly )
+        ch_assembly_contigs = FORMATSPADES.out.assembly
+        ch_versions    = ch_versions.mix(FORMATSPADES.out.versions)
+    } else if ( assembler == 'megahit' ) {
+        MEGAHIT_INTERLEAVED(
+            ch_pe_reads_to_assembly.toList(),
+            ch_se_reads_to_assembly.toList(),
+            'megahit_assembly'
+        )
+        MEGAHIT_INTERLEAVED.out.contigs
+            .map { [ [ id: 'megahit' ], it ] }
+            .set { ch_assembly_contigs }
+        ch_versions = ch_versions.mix(MEGAHIT_INTERLEAVED.out.versions)
+    } else { error 'Assembler not specified!' }
+
+    // If the user asked for length filtering, perform that with SEQTK_SEQ (the actual length parameter is used in modules.config)
+    if ( params.min_contig_length > 0 ) {
+        SEQTK_SEQ_CONTIG_FILTER ( ch_assembly_contigs )
+        ch_assembly_contigs = SEQTK_SEQ_CONTIG_FILTER.out.fastx
+        ch_versions = ch_versions.mix(SEQTK_SEQ_CONTIG_FILTER.out.versions)
+    }
+
+    //
+    // Call ORFs
+    //
+    ch_gff      = Channel.empty()
+    ch_protein  = Channel.empty()
+
+    //
+    // SUBWORKFLOW: Run PROKKA_SUBSETS on assmebly output, but split the fasta file in chunks of 10 MB, then concatenate and compress output.
+    //
+    if ( params.orf_caller == 'prokka' ) {
+        PROKKA_SUBSETS(ch_assembly_contigs, params.prokka_batchsize)
+        UNPIGZ_GFF(PROKKA_SUBSETS.out.gff.map { meta, gff -> [ [id: "${params.orf_caller}.${meta}"], gff ] })
+        ch_versions      = ch_versions.mix(PROKKA_SUBSETS.out.versions)
+        ch_gff           = UNPIGZ_GFF.out.unzipped
+        ch_protein       = PROKKA_SUBSETS.out.faa
+        ch_multiqc_files = ch_multiqc_files.mix(PROKKA_SUBSETS.out.prokka_log.collect{ meta, log -> log } )
```

Comment:
@erikrikarddaniel Check if this line is the cause of your problem. PROKKA_SUBSETS.out.prokka_log is `collect`ed in the workflow.

Reply:
You're referring to my Slack question? This is in line with what Jorge was into. Without having had time to look at it yet, I'm puzzled as to how this could pop up so suddenly after successfully running the pipeline many times. Thanks for the tip, anyway!

Reply:
I can test it right now. I will let you know ASAP